src/biz/workspace/publish_dup.rs

```diff
@@ -740,52 +744,82 @@ impl PublishCollabDuplicator {
       // assign a new id for the row
       let dup_row_id = gen_view_id();
       let mut db_row_collab = collab_from_doc_state(row_bin_data.clone(), &dup_row_id)?;
+      let mut db_row_body = DatabaseRowBody::open(pub_row_id.clone().into(), &mut db_row_collab)
+        .map_err(|e| AppError::Unhandled(e.to_string()))?;
 
       {
-        // update database_id and row_id in data
         let mut txn = db_row_collab.context.transact_mut();
-        let data = db_row_collab
-          .data
-          .get(&txn, "data")
-          .ok_or_else(|| {
-            AppError::RecordNotFound("no data found in database row collab".to_string())
-          })?
-          .cast::<MapRef>()
-          .map_err(|err| AppError::Unhandled(format!("data not map: {:?}", err)))?;
-        data.insert(&mut txn, "id", dup_row_id.clone());
-        data.insert(&mut txn, "database_id", new_db_id.clone());
+        // update database_id
+        db_row_body.update(&mut txn, |u| {
+          u.set_database_id(new_db_id.clone());
+        });
+
+        // get row document id before the id update
+        let pub_row_doc_id = db_row_body
+          .document_id(&txn)
+          .map_err(|e| AppError::Unhandled(e.to_string()))?;
+
+        // updates row id along with meta keys
+        db_row_body
+          .update_id(&mut txn, RowId::from(dup_row_id.clone()))
+          .map_err(|e| AppError::Unhandled(format!("failed to update row id: {:?}", e)))?;
+
+        // duplicate row document if exists
+        if let Some(pub_row_doc_id) = pub_row_doc_id {
+          if let Some(row_doc_doc_state) = published_db
+            .database_row_document_collabs
+            .get(&pub_row_doc_id)
+          {
+            let pub_doc_collab =
+              collab_from_doc_state(row_doc_doc_state.to_vec(), &pub_row_doc_id)?;
+            let pub_doc =
```

Comment:
if fail to open the doucment, can we just ignore its error and continuous the following tasks

Reply:
ok